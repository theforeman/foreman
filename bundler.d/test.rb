group :test do
  gem 'mocha', '~> 1.1', :require => false
  gem 'simplecov', '~> 0.9'
  gem 'spork-minitest', '0.0.3'
  gem 'single_test', '~> 0.6'
  gem 'minitest', '5.1.0'
  gem 'minitest-spec-rails' #, '~> 4.7'
  #gem 'ci_reporter', '2.0.0', :require => false
  gem 'ci_reporter_minitest', :require => false
  gem 'capybara', '~> 2.0'
  gem 'database_cleaner', '~> 1.3'
  gem 'launchy', '~> 2.4'
  gem 'spork', '~> 0.9'
  gem 'factory_girl_rails', '~> 4.5', :require => false
  gem 'rubocop-checkstyle_formatter', '~> 0.1'
  gem "poltergeist"
end
