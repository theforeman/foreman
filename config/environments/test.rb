# Track deprecation warnings in test environment as early as possible, but pause processing of
# deprecations until all plugins are registered (prior to the finisher_hook initializer) to ensure
# the whitelist is fully configured. This is done in the after_initialize block below.
ASDeprecationTracker.pause!

Foreman::Application.configure do
  # Settings specified here will take precedence over those in config/application.rb.

  # The test environment is used exclusively to run your application's
  # test suite. You never need to work with it otherwise. Remember that
  # your test database is "scratch space" for the test suite and is wiped
  # and recreated between test runs. Don't rely on the data there!
  config.cache_classes = true

  # Do not eager load code on boot. This avoids loading your whole application
  # just for the purpose of running a single test. If you are using a tool that
  # preloads Rails for running tests, you may have to set it to true.
  config.eager_load = false

  # Configure static asset server for tests with Cache-Control for performance.
  config.serve_static_files   = true
  config.static_cache_control = 'public, max-age=3600'

  # Adds additional error checking when serving assets at runtime.
  # Checks for improperly declared sprockets dependencies.
  # Raises helpful error messages.
  config.assets.raise_runtime_errors = true

  # Show full error reports and disable caching.
  config.consider_all_requests_local       = true
  config.action_controller.perform_caching = false

  # Raise exceptions instead of rendering exception templates.
  config.action_dispatch.show_exceptions = false

  # Disable request forgery protection in test environment.
  config.action_controller.allow_forgery_protection = false

  # Tell Action Mailer not to deliver emails to the real world.
  # The :test delivery method accumulates sent emails in the
  # ActionMailer::Base.deliveries array.
  config.action_mailer.delivery_method = :test
  config.action_mailer.perform_deliveries = true
  ActionMailer::Base.deliveries.clear

  # Use SQL instead of Active Record's schema dumper when creating the test database.
  # This is necessary if your schema can't be completely dumped by the schema dumper,
  # like if you have constraints or database-specific column types
  # config.active_record.schema_format = :sql

  # Should ANSI color codes be used when logging information
  config.colorize_logging = Foreman::Logging.config[:colorize]

  # Print deprecation notices to the stderr.
  config.active_support.deprecation = :stderr

  # Raise exception on mass assignment of unfiltered parameters
  config.action_controller.action_on_unpermitted_parameters = :strict

  # Use separate cache stores for parallel_tests
  config.cache_store = :file_store, Rails.root.join("tmp", "cache", "paralleltests#{ENV['TEST_ENV_NUMBER']}")

  # Enable automatic creation/migration of the test DB when running tests
  config.active_record.maintain_test_schema = true

  # Randomize the order test cases are executed.
  config.active_support.test_order = :random

  config.webpack.dev_server.enabled = false

  # Whitelist all plugin engines by default from raising errors on deprecation warnings for
  # compatibility, allow them to override it by adding an ASDT configuration file.
  config.after_initialize do
    Foreman::Plugin.all.each do |plugin|
      unless File.exist?(File.join(plugin.path, 'config', 'as_deprecation_whitelist.yaml'))
        ASDeprecationTracker.whitelist.add(engine: plugin.id.to_s.gsub('-', '_'))
      end
    end
    ASDeprecationTracker.resume!
  end
end
