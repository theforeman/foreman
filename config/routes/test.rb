if Rails.env.test?
  Foreman::Application.routes.draw do
    resources :testable, :only => :index
    resources :testable_strong_params, :only => :create
    resources :my_hosts, :only => :create
    resources :testable_no_params, :only => :create

    namespace :api do
      namespace :v2 do
        resources :testable, :only => [:create, :index]
      end

      resources :testable, :only => :index do
        get :raise_error, :on => :collection
      end
    end
  end
end
