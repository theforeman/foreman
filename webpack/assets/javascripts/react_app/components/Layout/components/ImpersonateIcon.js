import React from 'react';
import PropTypes from 'prop-types';

import { OverlayTrigger, Tooltip, Icon, MessageDialog } from 'patternfly-react';
import { translate as __ } from '../../../common/I18n';

import './ImpersonateIcon.scss';

class ImpersonateIcon extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      showModal: false,
    };
  }

  toggleModal = () => this.setState({ showModal: !this.state.showModal });

  stopImpersonating = url => () => {
    const stopUrl = `${window.location.origin}/${url}`;
    window.location.href = stopUrl;
  };

  render() {
    return (
      <React.Fragment>
        <OverlayTrigger
          overlay={
            <Tooltip id="stop-impersonation">
              {__(
                'You are impersonating another user, click to stop the impersonation'
              )}
            </Tooltip>
          }
          placement="bottom"
          trigger={['hover', 'focus']}
          rootClose={false}
        >
          <li className="drawer-pf-trigger masthead-icon">
            <span className="nav-item-iconic" onClick={this.toggleModal}>
              <Icon name="eye avatar small" className="blink-image" />
            </span>
          </li>
        </OverlayTrigger>
        <MessageDialog
          show={this.state.showModal}
          onHide={this.toggleModal}
          primaryAction={this.stopImpersonating(
            this.props.stopImpersonationUrl
          )}
          secondaryAction={this.toggleModal}
          primaryActionButtonContent={__('Confirm')}
          secondaryActionButtonContent={__('Cancel')}
          title={__('Confirm Action')}
          primaryContent={__(
            'You are about to stop impersonating other user. Are you sure?'
          )}
        />
      </React.Fragment>
    );
  }
}

ImpersonateIcon.propTypes = {
  stopImpersonationUrl: PropTypes.string.isRequired,
};

export default ImpersonateIcon;
