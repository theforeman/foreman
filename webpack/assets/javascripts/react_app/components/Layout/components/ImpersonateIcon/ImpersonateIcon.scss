@import "~@theforeman/vendor/scss/variables";

.masthead-icon {
  margin-top: 4px;
}

@keyframes blink {
  0% {
    opacity: 1;
  }

  50% {
    opacity: 0;
  }

  100% {
    opacity: 1;
  }
}

.blink-image {
  -moz-animation: blink normal 1s infinite ease-in-out; /* Firefox */
  -webkit-animation: blink normal 1s infinite ease-in-out; /* Webkit */
  -ms-animation: blink normal 1s infinite ease-in-out; /* IE */
  animation: blink normal 1s infinite ease-in-out; /* Opera and prob css3 final iteration */
}

.navbar-pf-vertical .nav {
  .nav-item-iconic .fa.blink-image {
    color: $color-pf-gold-200;
  }

  .nav-item-iconic:hover .fa.blink-image {
    color: $color-pf-gold-200;
  }
}
