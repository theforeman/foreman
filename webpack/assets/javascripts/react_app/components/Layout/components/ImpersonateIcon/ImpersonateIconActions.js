import { API } from '../../../../redux/API';
import { foremanUrl } from '../../../../../foreman_tools';

import { addToast } from '../../../ToastsList';

export const stopImpersonating = url => async dispatch => {
  try {
    const { data } = await API.delete(url);
    window.location.href = foremanUrl('/users');
    return dispatch(
      addToast({
        type: data.type,
        message: data.message,
      })
    );
  } catch (error) {
    return dispatch(
      addToast({
        type: 'error',
        message: 'Failed to stop impersonation',
      })
    );
  }
};
