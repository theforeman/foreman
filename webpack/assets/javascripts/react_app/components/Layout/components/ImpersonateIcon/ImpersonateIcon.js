import React, { useState } from 'react';
import PropTypes from 'prop-types';

import { OverlayTrigger, Tooltip, Icon, MessageDialog } from 'patternfly-react';
import { translate as __ } from '../../../../common/I18n';

import './ImpersonateIcon.scss';

const ImpersonateIcon = props => {
  const [showModal, setShowModal] = useState(false);

  const toggleModal = () => setShowModal(!showModal);

  return (
    <React.Fragment>
      <OverlayTrigger
        overlay={
          <Tooltip id="stop-impersonation">
            {__(
              'You are impersonating another user, click to stop the impersonation'
            )}
          </Tooltip>
        }
        placement="bottom"
        trigger={['hover', 'focus']}
        rootClose={false}
      >
        <li className="drawer-pf-trigger masthead-icon">
          <span className="nav-item-iconic" onClick={toggleModal}>
            <Icon name="eye avatar small" className="blink-image" />
          </span>
        </li>
      </OverlayTrigger>
      <MessageDialog
        show={showModal}
        onHide={toggleModal}
        primaryAction={() =>
          props.stopImpersonating(props.stopImpersonationUrl)
        }
        secondaryAction={toggleModal}
        primaryActionButtonContent={__('Confirm')}
        secondaryActionButtonContent={__('Cancel')}
        title={__('Confirm Action')}
        primaryContent={__(
          'You are about to stop impersonating other user. Are you sure?'
        )}
      />
    </React.Fragment>
  );
};

ImpersonateIcon.propTypes = {
  stopImpersonationUrl: PropTypes.string.isRequired,
  stopImpersonating: PropTypes.func.isRequired,
};

export default ImpersonateIcon;
