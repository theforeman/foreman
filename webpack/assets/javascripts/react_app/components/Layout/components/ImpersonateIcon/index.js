import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import ImpersonateIcon from './ImpersonateIcon';

import * as ImpersonateIconActions from './ImpersonateIconActions';

const mapDispatchToProps = dispatch =>
  bindActionCreators(ImpersonateIconActions, dispatch);

export default connect(null, mapDispatchToProps)(ImpersonateIcon);
