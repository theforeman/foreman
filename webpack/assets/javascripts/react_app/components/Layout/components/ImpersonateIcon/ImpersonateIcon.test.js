import { testComponentSnapshotsWithFixtures } from '@theforeman/test';

import ImpersonateIcon from './ImpersonateIcon';

const fixtures = {
  'should render': {
    stopImpersonationUrl: '/stop_impersonation',
    stopImpersonating: () => {},
  },
};

describe('ImpersonateIcon', () =>
  testComponentSnapshotsWithFixtures(ImpersonateIcon, fixtures));
