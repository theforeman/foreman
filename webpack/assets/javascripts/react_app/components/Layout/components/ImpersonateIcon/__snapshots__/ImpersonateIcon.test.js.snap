// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImpersonateIcon should render 1`] = `
<Fragment>
  <OverlayTrigger
    defaultOverlayShown={false}
    overlay={
      <Tooltip
        bsClass="tooltip"
        id="stop-impersonation"
        placement="right"
      >
        You are impersonating another user, click to stop the impersonation
      </Tooltip>
    }
    placement="bottom"
    rootClose={false}
    trigger={
      Array [
        "hover",
        "focus",
      ]
    }
  >
    <li
      className="drawer-pf-trigger masthead-icon"
    >
      <span
        className="nav-item-iconic"
        onClick={[Function]}
      >
        <Icon
          className="blink-image"
          name="eye avatar small"
          type="fa"
        />
      </span>
    </li>
  </OverlayTrigger>
  <MessageDialog
    accessibleDescription=""
    accessibleName=""
    className=""
    enforceFocus={true}
    footer={null}
    icon={null}
    onHide={[Function]}
    primaryAction={[Function]}
    primaryActionButtonBsStyle="primary"
    primaryActionButtonContent="Confirm"
    primaryContent="You are about to stop impersonating other user. Are you sure?"
    secondaryAction={[Function]}
    secondaryActionButtonBsStyle="default"
    secondaryActionButtonContent="Cancel"
    secondaryContent={null}
    show={false}
    title="Confirm Action"
  />
</Fragment>
`;
